### PR TITLE
Fix multiple flaky tests

### PR DESCRIPTION
**Pull request by alesapin**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Error: https://s3.amazonaws.com/clickhouse-test-reports/0/e426a9a2f18362ea69809b0affd2006fcc6ced09/stateless_tests__release__databaseordinary__actions_.html

### CODE REVIEW

File: tests/queries/0_stateless/01108_restart_replicas_rename_deadlock_zookeeper.sh [hunk 1]
```diff
@@ -38,27 +38,18 @@ function restart_replicas_loop()
...
-{
-    restart_replicas_loop
-}
 
 export -f rename_thread_1
 export -f rename_thread_2
-export -f restart_thread_1
-export -f restart_thread_2
+export -f restart_replicas_loop
+export -f restart_replicas_loop
```
Comment (azat) [inline tests/queries/0_stateless/01108_restart_replicas_rename_deadlock_zookeeper.sh:45]: Duplicate

```suggestion
```

File: tests/queries/0_stateless/00385_storage_file_and_clickhouse-local_app_long.sh [hunk 2]
```diff
@@ -19,11 +19,11 @@ function pack_unpack_compare()
 
     ${CLICKHOUSE_CLIENT} --query "CREATE TABLE buf_00385 ENGINE = Memory AS $1"
     local res_orig
-    res_orig=$(${CLICKHOUSE_CLIENT} --max_threads=1 --query "SELECT $TABLE_HASH FROM buf_00385")
+    res_orig=$(${CLICKHOUSE_CLIENT} --max_block_size=65505 --max_threads=1 --query "SELECT $TABLE_HASH FROM buf_00385")
```
Comment (azat) [inline tests/queries/0_stateless/00385_storage_file_and_clickhouse-local_app_long.sh:22]: Maybe it is better to use `# Tags: no-random-settings` ?

File: tests/queries/0_stateless/01268_procfs_metrics.sh [hunk 3]
```diff
@@ -41,4 +41,13 @@ timeout $TIMEOUT bash -c show_processes_func &
 
 wait
 
+# otherwise it can be alive after test
```
Comment (azat) [inline tests/queries/0_stateless/01268_procfs_metrics.sh:44]: It is better to:
- run `read_numbers_func` under `clickhouse_client_timeout`
- run `show_processes_func` under `clickhouse_client_loop_timeout`
That way client will wait the query.

Also you can use `clickhouse_test_wait_queries` at the end of the test to wait unfinished queries, but this should not be required.

Comment (alesapin) [reply tests/queries/0_stateless/01268_procfs_metrics.sh:44]: I'll merge this one and fix it in another PR.

File: tests/queries/0_stateless/00385_storage_file_and_clickhouse-local_app_long.sh (hunk 2 above)
Comment (azat) [reply tests/queries/0_stateless/00385_storage_file_and_clickhouse-local_app_long.sh:22]: Oh, I see, got the idea.

**Pull request merged**